cavalier_contours/src/core/math/line_line_intersect.rs

```diff
@@ -34,6 +34,47 @@ where
 
 /// Finds the intersects between two lines segments.
 ///
+/// This function returns the parametric solution(s) using the general line segment equation `P(t) = p0 + t * (p1 - p0)`.
+/// Note that this equation holds for both line segments.
+///
+/// # Explanation on result cases `LineLineIntr`
+/// ## `NoIntersect`
+/// Either of the following cases:
+/// * Lines are (almost) parallel
+/// * Both line segments are points and distinct from each other
+/// * One line segment is a point and distinct from the other line segment
+/// * Line segments are parallel and atleast one must be extended to intersect
+///
+/// ## `TrueIntersect`
+/// Either of the following cases:
+/// * Line segments are not parallel and intersect at one point
+/// * Both line segments are points and lie over each other
+/// * One line segment is a point and lies in other line segment
+///
+/// ## `FalseIntersect`
+/// Either of the following cases:
+/// * Line segments are not parallel and atleast one must be extended to intersect
+///
+/// ## `Overlapping`
+/// Either of the following cases:
+/// * Line segments are collinear and overlap
+///
+/// # Examples
+///
+/// ```
+/// # use cavalier_contours::core::traits::*;
+/// # use cavalier_contours::core::math::*;
+/// # use cavalier_contours::core::math::LineLineIntr::TrueIntersect;
+/// // A line segment tangent-intersecting a circle with one of the line segments end points.
```

Comment:
Copy paste of comment, it should describe the line-line intersect but was copied from the line-circle intersect doc.